### PR TITLE
Otto 0.2

### PR DESCRIPTION
**Pull request by mitchellh**
This is for later this week, but would love a review!

### CODE REVIEW

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+in the documentation.
+
+<a id="implicit">
+
+## Implicit Dependencies
+
+Otto requires _zero configuration_. It supports application configuration
+through an [Appfile](#),
+but the presence of this file is completely optional. If the file isn't
+found, Otto will inspect your environment and deterine a default configuration
```
Comment (armon) [inline source/blog/2016-01-07-otto-0-2.html.md]: Typo "determine"

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+As time goes on, Otto's detection of implicit dependencies will improve.
+For Otto 0.2, basic support for Ruby and Wordpress is available.
+Implicit dependencies can be easily disabled by creating a very simple
+Appfile.
+
+<a id="layered">
+
+## Layered Development Environments
+
+Otto 0.2 automatically layers development environments to vastly increase
```
Comment (armon) [inline source/blog/2016-01-07-otto-0-2.html.md]: This reads weird. Maybe "Otto 0.2 dramatically increases the setup speed of development environments by making use of intelligent layering."

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+software, etc. Otto 0.2 separates each of these steps out into layers to
+speed this up.
+
+For example, with a Ruby development environment, Ruby is compiled and
+installed in a layer. If you start another development environment with
+the same version of Ruby or you recreate that development environment,
+that entire step can be skipped unless you change the Ruby version.
+
+As a rough benchmark, on my machine building a Ruby environment went from
+10 minutes to 30 seconds.
```
Comment (armon) [inline source/blog/2016-01-07-otto-0-2.html.md]: Suggest "As an anecdote, on a 2015 Macbook Air building a Ruby ..."

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+We have plans to further speed up development environments in future versions
+of Otto, but feel that the speed is at a point where it isn't prohibitive to
+any workflow.
```
Comment (armon) [inline source/blog/2016-01-07-otto-0-2.html.md]: Suggest "While we plan to further optimize setup of development environments in future versions, they are currently at the point where focusing on other parts of Otto will be more fruitful."

**Comment by armon** [comment]
LGTM!

**Comment by mitchellh** [comment]
Good ideas, addressed your feedback.

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+As time goes on, Otto's detection of implicit dependencies will improve.
+For Otto 0.2, basic support for Ruby and Wordpress is available.
+Implicit dependencies can be easily disabled by creating a very simple
+Appfile.
+
+<a id="layered">
+
+## Layered Development Environments
+
+Otto 0.2 dramatically increases the time it takes to setup a development
```
Comment (armon) [inline source/blog/2016-01-07-otto-0-2.html.md]: I hope it doesn't dramatically increase (should be "decrease")

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    Compiling foundation: consul
+==> Compiling main application...
+==> Compilation success!
+</pre>
+
+In addition to the new application types, Otto 0.2 greatly improves almost every
+existing application type. The improvmeent highlights are listed below:
+
+  * Ruby now uses [ruby-install](#)
+    and [chruby](#)
```
Comment (squaresurf) [inline source/blog/2016-01-07-otto-0-2.html.md]: Did you mean to add the links for [chruby](https://github.com/postmodern/chruby) and [ruby-install](https://github.com/postmodern/ruby-install)?

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+Otto now has a plugin system for adding new application types.
+
+The plugin system is based on the same plugin system used by
+[Packer](https://packer.io) and [Terraform](https://terraform.io). Both
+of these projects have shown that even beginner-level Go developers find it
+simple to build new plugins. We hope that with such a plugin system in Otto,
+we lower the barrier to entry to contribute to Otto.
+
+In addition to contributing to the open source, the plugin system lets you
```
Comment (squaresurf) [inline source/blog/2016-01-07-otto-0-2.html.md]: Do you mean "contributing to ~~the~~ open source"?

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
+---
+page_title: "Otto 0.2"
+title: "Otto 0.2"
+list_image_url: "/images/blog/otto/logo.png"
+post_image_url: "/images/blog/otto/header.png"
+tags: otto
+author: Mitchell Hashimoto
+---
+
+We've released [Otto 0.2](https://ottoproject.io). Otto is a single
```
Comment (sethvargo) [inline source/blog/2016-01-07-otto-0-2.html.md]: Can you do a quick find-and-replace to make sure we always use https://www.PROJECT.io?

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+## New and Improved App Types
+
+Otto works best when it has a deep understanding of the environment it
+is being used in. Otto 0.2 ships with built-in support for new
+application types: Python, Java, and Rails. In addition to the new app types,
+every existing application type has been improved in some way.
+
+For Python applications, Otto will pre-install and configure Python,
+pip, and VirtaulEnv for your environment. You can customize the Python
```
Comment (sethvargo) [inline source/blog/2016-01-07-otto-0-2.html.md]: VirtualEnv

File: source/blog/2016-01-07-otto-0-2.html.md (hunk 6 above)
Comment (sethvargo) [inline source/blog/2016-01-07-otto-0-2.html.md]: I would say we don't link to these things. We don't link to Python, VirtualEnv, or Maven above.

Comment (squaresurf) [inline source/blog/2016-01-07-otto-0-2.html.md]: So they shouldn't be links then? I'm cool either way, I just noticed that they were links going nowhere.

File: source/blog/2016-01-07-otto-0-2.html.md (hunk 8 above)
Comment (mitchellh) [inline source/blog/2016-01-07-otto-0-2.html.md]: No problem, done in next push.

**Comment by mitchellh** [comment]
Latest round of feedback addressed.

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+    Infrastructure: aws (simple)
+
+    Compiling infra...
+    Compiling foundation: consul
+==> Compiling main application...
+==> Compilation success!
+</pre>
+
+In addition to the new application types, Otto 0.2 greatly improves almost every
+existing application type. The improvmeent highlights are listed below:
```
Comment (burzinpatel) [inline source/blog/2016-01-07-otto-0-2.html.md]: typo: improvement

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+## Implicit Dependencies
+
+Otto requires _zero configuration_. It supports application configuration
+through an [Appfile](https://www.ottoproject.io/docs/concepts/appfile.html),
+but the presence of this file is completely optional. If the file isn't
+found, Otto will inspect your environment and determine a default configuration
+automatically.
+
+Otto 0.2 extends this even further by introducing a feature known as
+"implicit dependencies." In addition to inspecting the environment to determine
```
Comment (burzinpatel) [inline source/blog/2016-01-07-otto-0-2.html.md]: ...known as "implicit dependencies".

File: source/blog/2016-01-07-otto-0-2.html.md [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+across multiple environments: installing common packages, compiling
+software, etc. Otto 0.2 separates each of these steps out into layers to
+speed this up.
+
+For example, with a Ruby development environment, Ruby is compiled and
+installed in a layer. If you start another development environment with
+the same version of Ruby or you recreate that development environment,
+that entire step can be skipped unless you change the Ruby version.
+
+As an anecdote, on a 2015 MacBook Air building a Ruby environent went from
```
Comment (burzinpatel) [inline source/blog/2016-01-07-otto-0-2.html.md]: Would 'rebuilding' or 'recreating' be more accurate instead of 'building'?

Comment (mitchellh) [inline source/blog/2016-01-07-otto-0-2.html.md]: I think its fine as-is but the rest of your comments are great!

Comment (burzinpatel) [inline source/blog/2016-01-07-otto-0-2.html.md]: Cool, LGTM.

**Pull request merged**